pkg/controller/migmigration/task.go

```diff
@@ -590,16 +600,25 @@ func (t *Task) getBothClusters() []*migapi.MigCluster {
 }
 
 // Get both source and destination clients.
-func (t *Task) getBothClients() ([]k8sclient.Client, error) {
-	list := []k8sclient.Client{}
-	for _, cluster := range t.getBothClusters() {
-		client, err := cluster.GetClient(t.Client)
-		if err != nil {
-			log.Trace(err)
-			return nil, err
-		}
-		list = append(list, client)
+func (t *Task) getBothClientsWithNamespaces() ([]k8sclient.Client, [][]string, error) {
```

Comment:
This seems like a worthwhile addition.  However, I'm not convinced it should replace `getBothClients()`.  Although, there are no more usages it may be better _functional decomposition_ and forward thinking to layer them.  That is, add `getBothClientsWithNamespaces()` but have the implementation _use_ `getBothClients()`.